jobs/integr8ly/ocp4/osd/cluster/create/Jenkinsfile

```diff
@@ -0,0 +1,158 @@
+                            """
+                        )
+                    }
+                }
+            } // stage
+            
+            stage("Begin cluster creation") {
+                rootWorkspaceDir = sh (
+                    script: "pwd",
+                    returnStdout: true
+                ).trim()
+                installConfigDirectory = rootWorkspaceDir + "/rhmi-jenkins/" + clusterInfoDirectory + "/"
+                sh "ocm post /api/clusters_mgmt/v1/clusters --body=${installConfigDirectory}cluster.json > ${clusterInfoFileName}"
+                clusterID = sh (
+                    script: "cat ${clusterInfoFileName} | jq -r .id",
+                    returnStdout: true
+                ).trim()
+            } // stage
+
+            stage("Check if the cluster is ready") {
+                Boolean isReady = isClusterReady(clusterID)
+                while (!isReady) {
+                    println "Cluster is not ready yet"
+                    sleep time: 5, unit: 'MINUTES'
+                    isReady = isClusterReady(clusterID)
+                }
+            } // stage
+
+            stage("Obtain and persist cluster's credentials") {
+                Boolean areCredentialsAvailable = credentialsAvailable(clusterID)
+                while (!areCredentialsAvailable) {
+                    println "Cluster credentials not available yet"
+                    sleep time: 3, unit: 'MINUTES'
+                    areCredentialsAvailable = credentialsAvailable(clusterID)
+                }
+                sh "ocm get /api/clusters_mgmt/v1/clusters/${clusterID}/credentials > ${installConfigDirectory}${clusterCredentialsFileName}"
+                withCredentials([string(credentialsId: "gitlab-access-token", variable: "GITLAB_ACCESS_TOKEN")]) {
+                    dir("rhmi-jenkins") {
+                        sh(
+                            returnStdout: false,
+                            script: """
+                                git pull jenkins master
+                                git add -A
+                                git commit -m 'cluster created: ${clusterName}'
+                                git push jenkins master
+                            """
+                        )
+                    }
+                }
+            } // stage
+
+            stage("Logout from ocm") {
+                sh "ocm logout"
+            } // stage
+            
+        } catch (caughtError){
+            currentBuild.result = "FAILURE"
+            err = caughtError
+            println caughtError
+        } finally {
+            notifyBuild(currentBuild.result, err)
+        }
+    }}} // timeout, ansiColor, timestamps
+} // node
+
+def isClusterReady(clusterID) {
+    String clusterStatus = sh (
+        script: "ocm get /api/clusters_mgmt/v1/clusters/${clusterID}/status | jq -r .state",
+        returnStdout: true
+    ).trim()
+    return clusterStatus.contains("ready")
```

Comment:
Just out of curiosity - what does it return when it's not ready?

Reply:
`installing`

Reply:
OK thanks